scripts/build.py

```diff
@@ -1190,6 +1206,8 @@ def compile_typescript_files(project_dir):
         project_dir: str. The project directory which contains the ts files
             to be compiled.
     """
+    test_compiled_js_dir_is_valid()
+    safe_delete_directory_tree(COMPILED_JS_DIR)
```

Comment:
Just to check: if we delete the compiled js dir, does that result in any kind of slowdown? If so, regenerating everything might be an issue in "continuous run" mode, and I'm wondering if in that case there's a more fine-grained criterion we should use in order to figure out when to regenerate.

Reply:
No, it does not. We are only deleting when we are doing a fresh compilation. When a watcher is running, there is no deletion but if the user shuts down the server and the compilation is triggered again, all files will be recompiled irrespective of whether the old files are present or not.

Reply:
OK, sounds good. Thanks!